src/euroeval/benchmark_modules/litellm.py

```diff
@@ -352,24 +365,38 @@ def generate(self, inputs: dict) -> GenerativeModelOutput:
         if self.buffer["first_label_token_mapping"]:
             generation_kwargs["logprobs"] = True
             generation_kwargs["top_logprobs"] = MAX_LOGPROBS
-        if self.model_config.revision == "thinking":
-            generation_kwargs["thinking"] = dict(
+
+        param = ""
+
+        if self.model_config.parameter:
+            param = self.model_config.parameter
+        elif self.model_config.revision in ALLOWED_PARAMS.get(self.model_config.model_id, []):
+            import warnings
+            warnings.warn(
+        f"Using '@' for parameters (like '{self.model_config.revision}') is deprecated. "
+        f"Please use '#' instead (e.g., '{self.model_config.model_id}#{self.model_config.revision}').",
+        DeprecationWarning,
+        stacklevel=2
+    )
+            param = self.model_config.revision
```

Comment:
This is the second occurrence of this block, regarding extracting the parameter. I would instead create a separate method, `_get_param`, which does this, to avoid duplication.